packages/config/src/VitessceAutoConfig.js

```diff
@@ -0,0 +1,384 @@
+      .filter(key => key.toLowerCase().includes('cluster') || key.toLowerCase().includes('cell_type'));
+
+    if (hasCellSetData.length > 0) {
+      views.push(['obsSets']);
+    }
+
+    this.metadataSummary.obsm.forEach((key) => {
+      if (key.toLowerCase().includes('obsm/x_umap')) {
+        views.push(['scatterplot', { mapping: 'UMAP' }]);
+      }
+      if (key.toLowerCase().includes('obsm/x_tsne')) {
+        views.push(['scatterplot', { mapping: 't-SNE' }]);
+      }
+      if (key.toLowerCase().includes('obsm/x_pca')) {
+        views.push(['scatterplot', { mapping: 'PCA' }]);
+      }
+      if (key.toLowerCase().includes(('obsm/x_segmentations'))) {
+        views.push(['layerController']);
+      }
+      if (key.toLowerCase().includes(('obsm/x_spatial'))) {
+        views.push(['spatial']);
+      }
+    });
+
+    if (this.metadataSummary.X) {
+      views.push(['heatmap']);
+      views.push(['featureList']);
+    }
+
+    return views;
+  }
+
+  async setMetadataSummary() {
+    if (Object.keys(this.metadataSummary).length > 0) {
+      return this.metadataSummary;
+    }
+
+    const parseMetadataFile = (metadataFile) => {
+
+      if (!metadataFile.metadata) {
+        throw new Error('Could not generate config: .zmetadata file is not valid.');
+      }
+
+      const obsmKeys = Object.keys(metadataFile.metadata)
+        .filter(key => key.startsWith('obsm/X_'))
+        .map(key => key.split('/.zarray')[0]);
+
+      const obsKeysArr = Object.keys(metadataFile.metadata)
+        .filter(key => key.startsWith('obs/') && !key.includes('obs/.') && !key.includes('obs/__'))
+        .map(key => key.split('/.za')[0]);
+
+      function uniq(a) {
+        return a.sort().filter((item, pos, ary) => !pos || item !== ary[pos - 1]);
+      }
+      const obsKeys = uniq(obsKeysArr);
+
+      const X = Object.keys(metadataFile.metadata).filter(key => key.startsWith('X'));
+
+      const out = {
+        obsm: obsmKeys,
+        obs: obsKeys,
+        X: X.length > 0,
+      };
+
+      return out;
+    };
+
+    const metadataExtension = '.zmetadata';
+    const url = [this.fileUrl, metadataExtension].join('/');
+    return fetch(url).then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      return Promise.reject(response);
+    })
+      .then(responseJson => parseMetadataFile(responseJson))
+      .catch((error) => {
+        if (error.status === 404) {
+          const errorMssg = `Could not generate config. File ${metadataExtension} not found in supplied file URL. Check docs for more explanation.`;
+          return Promise.reject(new Error(errorMssg));
+        }
+        return Promise.reject(error);
+      });
+  }
+}
+
+const configClasses = [
+  {
+    extensions: ['ome.tif', '.ome.tiff', '.ome.tf2', '.ome.tf8'], // todo: test that ".ome.tf2", ".ome.tf8" work
+    class: OmeTiffAutoConfig,
+  },
+  {
+    extensions: ['h5ad.zarr', '.adata.zarr', '.anndata.zarr'],
```

Comment:
```suggestion
    extensions: ['.h5ad.zarr', '.adata.zarr', '.anndata.zarr'],
```

(Just to be consistent about whether the extensions begin with a period or not)